astropy/config/paths.py

```diff
@@ -243,16 +243,16 @@ def __enter__(self):
         # Special case for the config case, where we need to reset all the
         # cached config objects
         from .configuration import _cfgobjs
-
-        path = super().__enter__()
+        self._cfgobjs_copy = _cfgobjs.copy()
```

Comment:
The changes here are to *keep* the configuration instead of deleting it at the end so that it gets reloaded from the configuration file. This is because some config items can have been set programmatically (in particular, `utils.iers.conf.auto_download`)

Reply:
Is the "reset all the cached config objects" comment on L243 above still accurate? If not, should remove/modify it.

And this is a subtle API change for `set_temp_config`, right? I wonder if it warrants a change log.

Reply:
Yes, I guess you are right. I now updated the comment and added a changelog entry.